src/main/java/pages/HomePage.java

```diff
@@ -48,13 +49,14 @@ public void searchText(String textForSearch) {
         searchButton.click();
     }
 
-    public void clickOnAmazonLogo() {
-        amazonLogo.click();
-    }
-
-    public void clickOnTodaySDeals(String text) {
-        clickOnAmazonLogo();
-        driver.findElement(By.linkText(text)).click();
+    public void clickOnTodaySDeals() {
+        try {
+            if (driver.findElement(By.cssSelector("div[data-toaster-slot='DEFAULT']")).isDisplayed()) {
+                driver.findElement(By.xpath("//div[@class='glow-toaster-footer']/span[contains(@class,'dismiss')]")).click();
```

Comment:
Why don't you use PageFactory for elements? Why don't you use @FindBy annotations?